flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/dataformat/DataFormatConverters.java

```diff
@@ -0,0 +1,1022 @@
+
+		private BinaryStringConverter() {}
+
+		@Override
+		BinaryString toExternalImpl(BaseRow row, int column) {
+			return row.getString(column);
+		}
+	}
+
+	/**
+	 * Converter for BinaryArray.
+	 */
+	public static class BinaryArrayConverter extends IdentityConverter<BinaryArray> {
+
+		public static final BinaryArrayConverter INSTANCE = new BinaryArrayConverter();
+
+		private BinaryArrayConverter() {}
+
+		@Override
+		BinaryArray toExternalImpl(BaseRow row, int column) {
+			return row.getArray(column);
+		}
+	}
+
+	/**
+	 * Converter for BinaryMap.
+	 */
+	public static class BinaryMapConverter extends IdentityConverter<BinaryMap> {
+
+		public static final BinaryMapConverter INSTANCE = new BinaryMapConverter();
+
+		private BinaryMapConverter() {}
+
+		@Override
+		BinaryMap toExternalImpl(BaseRow row, int column) {
+			return row.getMap(column);
+		}
+	}
+
+	/**
+	 * Converter for String.
+	 */
+	public static class StringConverter extends DataFormatConverter<BinaryString, String> {
+
+		public static final StringConverter INSTANCE = new StringConverter();
+
+		private StringConverter() {}
+
+		@Override
+		BinaryString toInternalImpl(String value) {
+			return BinaryString.fromString(value);
+		}
+
+		@Override
+		String toExternalImpl(BinaryString value) {
+			return value.toString();
+		}
+
+		@Override
+		String toExternalImpl(BaseRow row, int column) {
+			return row.getString(column).toString();
+		}
+	}
+
+	public static final TimeZone LOCAL_TZ = TimeZone.getDefault();
+	public static final long MILLIS_PER_DAY = 86400000L; // = 24 * 60 * 60 * 1000
+
+	/** Converts the Java type used for UDF parameters of SQL TIME type
+	 * ({@link java.sql.Time}) to internal representation (int).
+	 *
+	 * <p>Converse of {@link #internalToTime(int)}. */
+	public static int toInt(java.sql.Time v) {
+		return (int) (toLong(v) % MILLIS_PER_DAY);
+	}
+
+	/** Converts the Java type used for UDF parameters of SQL DATE type
+	 * ({@link java.sql.Date}) to internal representation (int).
+	 *
+	 * <p>Converse of {@link #internalToDate(int)}. */
+	public static int toInt(java.util.Date v) {
+		return toInt(v, LOCAL_TZ);
+	}
+
+	public static int toInt(java.util.Date v, TimeZone timeZone) {
+		return (int) (toLong(v, timeZone) / MILLIS_PER_DAY);
+	}
+
+	public static long toLong(java.util.Date v) {
+		return toLong(v, LOCAL_TZ);
+	}
+
+	/** Converts the Java type used for UDF parameters of SQL TIMESTAMP type
+	 * ({@link java.sql.Timestamp}) to internal representation (long).
+	 *
+	 * <p>Converse of {@link #internalToTimestamp(long)}. */
+	public static long toLong(Timestamp v) {
+		return toLong(v, LOCAL_TZ);
+	}
+
+	public static long toLong(java.util.Date v, TimeZone timeZone) {
+		long time = v.getTime();
+		return time + (long) timeZone.getOffset(time);
+	}
+
+	/** Converts the internal representation of a SQL DATE (int) to the Java
+	 * type used for UDF parameters ({@link java.sql.Date}). */
+	public static java.sql.Date internalToDate(int v) {
```

Comment:
Could we move the date time relative utilities to a specific place, such as `org.apache.flink.table.runtime.functions.DateTimeUtils`.

Currently, it is a bit messy putting date time functions in various classes.